### PR TITLE
Consolidate containerd config.toml in single file

### PR DESCRIPTION
**Pull request by primeroz**
This PR:

- Consolidate the full `config.toml` configuration file for containerd into a single file to address [#1737](https://github.com/giantswarm/roadmap/issues/1737)

@erkanerol as discussed in https://gigantic.slack.com/archives/C01BYMF6RN0/p1689847741604689?thread_ts=1689844311.988129&cid=C01BYMF6RN0 can you verify it when using the `credentials and mirrors` configurations ? the structure of values has not changed for those settings

### Testing

Description on how {APP-NAME} can be tested.

- [x] fresh install works
- [x] upgrade from previous version works

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [ ] Update `/examples` if required.

### CODE REVIEW

**Comment by primeroz** [comment]
Local rendering diff between `main` and this `branch`

```diff
@@ -1,4 +1,12 @@
 ---
+# Source: cluster-cloud-director/templates/registry-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: release-name-registry-configuration-0f196508
+data:
+  registry-config.toml: "dmVyc2lvbiA9IDIKCiMgcmVjb21tZW5kZWQgZGVmYXVsdHMgZnJvbSBodHRwczovL2dpdGh1Yi5jb20vY29udGFpbmVyZC9jb250YWluZXJkL2Jsb2IvbWFpbi9kb2NzL29wcy5tZCNiYXNlLWNvbmZpZ3VyYXRpb24KIyBzZXQgY29udGFpbmVyZCBhcyBhIHN1YnJlYXBlciBvbiBsaW51eCB3aGVuIGl0IGlzIG5vdCBydW5uaW5nIGFzIFBJRCAxCnN1YnJlYXBlciA9IHRydWUKIyBzZXQgY29udGFpbmVyZCdzIE9PTSBzY29yZQpvb21fc2NvcmUgPSAtOTk5CmRpc2FibGVkX3BsdWdpbnMgPSBbXQpbcGx1Z2lucy4iY29udGFpbmVyZC5ydW50aW1lLnYxLmxpbnV4Il0KIyBzaGltIGJpbmFyeSBuYW1lL3BhdGgKc2hpbSA9ICJjb250YWluZXJkLXNoaW0iCiMgcnVudGltZSBiaW5hcnkgbmFtZS9wYXRoCnJ1bnRpbWUgPSAicnVuYyIKIyBkbyBub3QgdXNlIGEgc2hpbSB3aGVuIHN0YXJ0aW5nIGNvbnRhaW5lcnMsIHNhdmVzIG9uIG1lbW9yeSBidXQKIyBsaXZlIHJlc3RvcmUgaXMgbm90IHN1cHBvcnRlZApub19zaGltID0gZmFsc2UKCltwbHVnaW5zLiJpby5jb250YWluZXJkLmdycGMudjEuY3JpIi5jb250YWluZXJkLnJ1bnRpbWVzLnJ1bmNdCiMgc2V0dGluZyBydW5jLm9wdGlvbnMgdW5zZXRzIHBhcmVudCBzZXR0aW5ncwpydW50aW1lX3R5cGUgPSAiaW8uY29udGFpbmVyZC5ydW5jLnYyIgpbcGx1Z2lucy4iaW8uY29udGFpbmVyZC5ncnBjLnYxLmNyaSIuY29udGFpbmVyZC5ydW50aW1lcy5ydW5jLm9wdGlvbnNdClN5c3RlbWRDZ3JvdXAgPSB0cnVlCltwbHVnaW5zLiJpby5jb250YWluZXJkLmdycGMudjEuY3JpIl0Kc2FuZGJveF9pbWFnZSA9ICJwcm9qZWN0cy5yZWdpc3RyeS52bXdhcmUuY29tLy90a2cvcGF1c2U6My43IgoKW3BsdWdpbnMuImlvLmNvbnRhaW5lcmQuZ3JwYy52MS5jcmkiLnJlZ2lzdHJ5XQogIFtwbHVnaW5zLiJpby5jb250YWluZXJkLmdycGMudjEuY3JpIi5yZWdpc3RyeS5taXJyb3JzXQpbcGx1Z2lucy4iaW8uY29udGFpbmVyZC5ncnBjLnYxLmNyaSIucmVnaXN0cnkuY29uZmlnc10KICAK"
+---
 # Source: cluster-cloud-director/templates/cluster.yaml
 apiVersion: cluster.x-k8s.io/v1beta1
 kind: Cluster
@@ -287,7 +295,7 @@
     application.giantswarm.io/team: "rocket"
     app.kubernetes.io/version: "0.12.2"
     helm.sh/chart: "cluster-cloud-director-0.12.2"
-  name: release-name-worker-57df471b
+  name: release-name-worker-cf065c25
   namespace: org-giantswarm
 spec:
   template:
@@ -335,7 +343,12 @@
             AllowTcpForwarding no
             AllowAgentForwarding no
             
-        
+        - path: /etc/containerd/config.toml
+          permissions: "0600"
+          contentFrom:
+            secret:
+              name: release-name-registry-configuration-0f196508
+              key: registry-config.toml
       
       preKubeadmCommands:
       - /bin/test ! -d /var/lib/kubelet && (/bin/mkdir -p /var/lib/kubelet && /bin/chmod 0750 /var/lib/kubelet)
@@ -523,7 +536,12 @@
           sed -i 's/nodeRegistration:/skipPhases: ["addon\/coredns", "addon\/kube-proxy"]\nnodeRegistration:/' "${dir}/kubeadm.yaml"
           { set +x; } 2>/dev/null
           
-      
+      - path: /etc/containerd/config.toml
+        permissions: "0600"
+        contentFrom:
+          secret:
+            name: release-name-registry-configuration-0f196508
+            key: registry-config.toml
       - path: /tmp/kubeadm/patches/kube-apiserver+json.tpl
         content: |-
           [
@@ -653,7 +671,7 @@
         configRef:
           apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
           kind: KubeadmConfigTemplate
-          name: release-name-worker-57df471b
+          name: release-name-worker-cf065c25
           namespace: org-giantswarm
       clusterName: release-name
       infrastructureRef:
```

**Review by erkanerol — APPROVED** [review]

**Comment by erkanerol** [comment]
Let's wait for https://github.com/giantswarm/cluster-cloud-director/pull/202 and then merge.

File: helm/cluster-cloud-director/README.md [hunk 1]
```diff
@@ -65,7 +69,7 @@ Configurations related to cluster connectivity such as container registries.
 | `connectivity.ntp.servers[*]` | **Server**|**Type:** `string`<br/>|
 | `connectivity.proxy` | **Proxy** - Whether/how outgoing traffic is routed through proxy servers.|**Type:** `object`<br/>|
 | `connectivity.proxy.enabled` | **Enable**|**Type:** `boolean`<br/>|
-| `connectivity.proxy.secretName` | **Secret name** - Name of a secret resource used by containerd to obtain the HTTP_PROXY, HTTPS_PROXY, and NO_PROXY environment variables.|**Type:** `string`<br/>**Value pattern:** `^[a-z0-9-]{1,63}$`<br/>|
+| `connectivity.proxy.secretName` | **Secret name** - Name of a secret resource used by containerd to obtain the HTTP_PROXY, HTTPS_PROXY, and NO_PROXY environment variables. If empty the value will be defaulted to <clusterName>-cluster-values.|**Type:** `string`<br/>**Value pattern:** `^[a-z0-9-]{0,63}$`<br/>|
```
Comment (jkremser) [inline helm/cluster-cloud-director/README.md:72]:
```suggestion
| `connectivity.proxy.secretName` | **Secret name** - Name of a secret resource used by containerd to obtain the HTTP_PROXY, HTTPS_PROXY, and NO_PROXY environment variables. If empty, the value will be defaulted to <clusterName>-cluster-values.|**Type:** `string`<br/>**Value pattern:** `^[a-z0-9-]{0,63}$`<br/>|
```

Comment (jkremser) [reply helm/cluster-cloud-director/README.md:72]: nit: missing comma

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Set value for `controller-manager` `terminated-pod-gc-threshold` to `125` ( consistent with vintage ) 
 
+### Changed
+- Consolidate containerd `config.toml` into single file to address [#1737](https://github.com/giantswarm/roadmap/issues/1737)
```
Comment (jkremser) [inline CHANGELOG.md:19]:
```suggestion
- Consolidate containerd `config.toml` into a single file to address [#1737](https://github.com/giantswarm/roadmap/issues/1737)
```
nit: typo

File: helm/cluster-cloud-director/values.schema.json [hunk 3]
```diff
@@ -671,6 +672,27 @@
                     "title": "Runtime defined entity (RDE) identifier",
                     "description": "This cluster's RDE ID in the VCD API."
                 },
+                "sandboxContainerImage": {
+                    "type": "object",
+                    "title": "Sandbox Container image",
```
Comment (jkremser) [inline helm/cluster-cloud-director/values.schema.json:677]:
```suggestion
                    "title": "Sandbox Container image (pause container)",
```

**Review by jkremser — APPROVED** [review]

**Pull request merged**